torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -319,41 +325,54 @@ std::shared_ptr<FutureMessage> ProcessGroupAgent::send(
   return future;
 }
 
+void ProcessGroupAgent::handleSend(SendWork work) {
```

Comment:
Can this be const SendWork& ?

as-is, this is going to require excess copying in SendWork (mostly the message.payload is an issue, tensor pay will be fine) - can you fix this before landing?

(folks might suggest std::move() to SendWork, though it doesn't look trivially possible from the caller; const ref is probably easiest/best here)

Reply:
Thanks for catching this!